docs/js-web-tools/grunt-gulp/grunt-gulp.rst

```diff
@@ -0,0 +1,388 @@
+
+The concat task creates a new combined.js file and places it into the temp directory. The jshint task simply runs and doesn’t produce output. The uglify task creates a new combined.min.js file and places it into wwwroot\lib. On completion, the solution should look something like the screenshot below:
+
+.. image:: _static/solution-explorer-after-all-tasks.png
+
+.. note:: For more information on the options for each package, visit https://www.npmjs.com/ and lookup the package name in the search box on the main page. For example, you can look up the grunt-contrib-clean package to get a documentation link that explains all of its parameters.
+
+All Together Now
+----------------
+
+Use the Grunt **registerTask()** method to run a series of tasks in a particular sequence. For example, to run the example steps above in the order clean -> concat -> jshint -> uglify, add the code below to the module. The code should be added to the same level as the loadNpmTasks() calls, outside initConfig.
+
+.. code-block:: javascript
+
+	grunt.registerTask("all", ['clean', 'concat', 'jshint', 'uglify']);
+
+The new task shows up in Task Runner Explorer under Alias Tasks. You can right-click and run it just as you would other tasks. The **all** task will run **clean, concat, jshint and uglify**, in order. 
+
+.. image:: _static/alias-tasks.png
+
+Watching For Changes
+--------------------
+
+A **watch** task keeps an eye on files and directories. The watch triggers tasks automatically if it detects changes. Add the code below to initConfig to watch for changes to *.js files in the TypeScript directory. If a JavaScript file is changed, **watch** will run the **all** task.
+
+.. code-block:: javascript
+
+	watch: {
+		files: ["TypeScript/*.js"],
+		tasks: ["all"]
+	}
+
+Add a call to **loadNpmTasks()** to show the **watch** task in Task Runner Explorer. 
+
+.. code-block:: javascript
+
+	grunt.loadNpmTasks('grunt-contrib-watch');
+
+Right-click the watch task in Task Runner Explorer and select Run from the context menu. The command window that shows the watch task running will display a waiting… message. Open one of the TypeScript files, add a space, and then save the file. This will trigger the watch task and trigger the other tasks to run in order. The screenshot below shows a sample run.
+
+.. image:: _static/watch-running.png
+
+Binding to Visual Studio Events
+-------------------------------
+
+Unless you want to manually start your tasks every time you work in Visual Studio, you can bind tasks to **Before Build**, **After Build**, **Clean**, and **Project Open** events. 
+
+Let’s bind **watch** so that it runs every time Visual Studio opens. In Task Runner Explorer, right-click the watch task and select **Bindings > Project Open** from the context menu. 
+
+.. image:: _static/bindings-project-open.png
+
+Unload and reload the project. When the project loads again, the watch task will start running automatically.
+
+Using Gulp
```

Comment:
We may want to split the article into 2 at that point. Perhaps file an issue after this article is accepted so we can close this PR?